packages/fluentui/react-northstar/src/components/Tree/Tree.tsx

```diff
@@ -245,186 +254,188 @@ class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
     const { id } = treeItemProps;
-    const { exclusive, items } = this.props;
+
     const siblings = getSiblings(items, id);
 
     const activeItemIdIndex = activeItemIds.indexOf(id);
+    let nextActiveItemsIds = activeItemIds;
 
     if (activeItemIdIndex > -1) {
-      activeItemIds = removeItemAtIndex(activeItemIds, activeItemIdIndex);
+      nextActiveItemsIds = removeItemAtIndex(activeItemIds, activeItemIdIndex);
     } else {
       if (exclusive) {
         siblings.some(sibling => {
           const activeSiblingIdIndex = activeItemIds.indexOf(sibling['id']);
           if (activeSiblingIdIndex > -1) {
-            activeItemIds = removeItemAtIndex(activeItemIds, activeSiblingIdIndex);
+            nextActiveItemsIds = removeItemAtIndex(activeItemIds, activeSiblingIdIndex);
 
             return true;
           }
           return false;
         });
       }
 
-      activeItemIds = [...activeItemIds, id];
+      nextActiveItemsIds = [...nextActiveItemsIds, id];
     }
 
-    this.setActiveItemIds(e, activeItemIds);
-  }
+    setActiveItemIds(e, nextActiveItemsIds);
+  };
 
-  onFocusFirstChild = (itemId: string) => {
-    const currentElement = this.itemsRef.get(itemId);
+  const onFocusFirstChild = React.useCallback((itemId: string) => {
+    const currentElement = itemsRef.get(itemId);
 
     if (!currentElement || !currentElement.current) {
       return;
     }
 
-    const elementToBeFocused = getNextElement(this.treeRef.current, currentElement.current);
+    const elementToBeFocused = getNextElement(treeRef.current, currentElement.current);
 
     if (!elementToBeFocused) {
       return;
     }
 
     elementToBeFocused.focus();
-  };
+  }, []);
 
-  onSiblingsExpand = (e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
-    const { exclusive, items } = this.props;
+  const onSiblingsExpand = React.useCallback((e: React.SyntheticEvent, treeItemProps: TreeItemProps) => {
     if (exclusive) {
       return;
     }
 
     const { id } = treeItemProps;
-    const { activeItemIds } = this.state;
+
     const siblings = getSiblings(items, id);
 
+    const nextActiveItemsIds = [...activeItemIds];
+
     siblings.forEach(sibling => {
-      if (hasSubtree(sibling) && !this.isActiveItem(sibling['id'])) {
-        activeItemIds.push(sibling['id']);
+      if (hasSubtree(sibling) && !isActiveItem(sibling['id'])) {
+        nextActiveItemsIds.push(sibling['id']);
       }
     });
 
-    if (hasSubtree(treeItemProps) && !this.isActiveItem(id)) {
-      activeItemIds.push(id);
+    if (hasSubtree(treeItemProps) && !isActiveItem(id)) {
+      nextActiveItemsIds.push(id);
     }
 
-    this.setActiveItemIds(e, activeItemIds);
-  };
-
-  setActiveItemIds = (e: React.SyntheticEvent, activeItemIds: string[]) => {
-    _.invoke(this.props, 'onActiveItemIdsChange', e, { ...this.props, activeItemIds });
-
-    this.setState({
-      activeItemIds,
-    });
-  };
+    setActiveItemIds(e, nextActiveItemsIds);
+  }, []);
```

Comment:
```suggestion
  }, [items, setActiveItemIds]);
```

There variables are dependencies. `items` and `setActiveItemIds` are unstable, oops